src/main/dynamo/UnprocessedKeyCollector.ts

```diff
@@ -0,0 +1,36 @@
+import { DynamoDB } from "aws-sdk"
+import { PartitionAndSortKey } from "./keys"
+import { Table } from "./Table"
+import { TaggedModel } from "./types"
+
+/** Collects a set of unprocessed DynamoDB keys returned from an operation (e.g. batchGet) and maps the "raw"
```

Comment:
Since both batchGet and batchWrite can return unprocessed keys, this class consolidates the logic of mapping raw unprocessed DynamoDB keys back to the beyonce keys that the caller originally passed in.

The "raw" keys are JSON and the Beyonce keys are class instances that carry around type information about which model they correspond to.